Data/NugetPackageSources.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+
+namespace FuGetGallery
+{
+    public static class NugetPackageSources
+    {
+        private static IConfiguration _configuration;
+
+        private static readonly Lazy<IReadOnlyList<NugetPackageSource>> _packageSources
+            = new Lazy<IReadOnlyList<NugetPackageSource>> (GetPackageSources);
+
+        public static void Configure (IConfiguration configuration)
+            => _configuration = configuration;
+
+        public static IReadOnlyList<NugetPackageSource> PackageSources => _packageSources.Value;
+
+        private static List<NugetPackageSource> GetPackageSources ()
```

Comment:
Hey, I'm a developer from the team that runs nuget.org. NuGet repositories are allowed to change the path of their resources at any time as long as they update their [service index](https://docs.microsoft.com/en-us/nuget/api/overview#service-index) (like `https://api.nuget.org/v3/index.json`). I would recommend following the NuGet protocol]: the configuration should accept a service index url, and should use that to discover the desired resources. For an example implementation, take a look at [`ServiceIndexService`](https://github.com/loic-sharma/BaGet/blob/master/src/BaGet.Protocol/Services/ServiceIndexService.cs). For the full reference on the service index, see: https://docs.microsoft.com/en-us/nuget/api/service-index

Reply:
That is definitely a great idea.  One problem is our internal implementation of Nuget Gallery doesn't have a service index.  Since it's internal, we really didn't need one, simply pointing VS at host/api/v2 was enough.

Also, I feel like this is a larger overhaul since the hard coded location pointed to nuget endpoints that return data in a different structure than anything from our internal nuget gallery.  For example, it uses "https://api-v2v3search-0.nuget.org/query?prerelease=true&q={0}" and there is nothing in the base Nuget Gallery that I could find to return that exact structure.  I may have missed something though.

My hope is this initial PR will at least make those points configurable, and then a future enhancement could support service index discoverability.  Thoughts?

Reply:
I suppose it's up to @praeclarum. I would advocate using the service index as this will support more upstream services.